internal/middleware/persistctx.go

```diff
@@ -0,0 +1,26 @@
+package middleware
+
+import (
+	"codeberg.org/gruf/go-pctx"
+	"github.com/gin-gonic/gin"
+)
+
+// PersistContext is a middleware handler that replaces an existing
+// request context with one capable of storing persisting key-values.
+// Ideally this should be the first middleware in the handler stack.
+func PersistContext() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		// Get base request context.
+		base := ctx.Request.Context()
+
+		// Wrap within persistent ctx.
+		pCtx := pctx.Get(base)
```

Comment:
I understand what the code does having looked at the underlying library, but I'm not sure I understand what benefits it brings?

Reply:
this part isn't strictly necessary right now, so I could remove it. but it's useful if we say start storing information in the context further down the stack, but want it accessible in the logging hooks.

though again I realize it may not be useful right now so can remove it if you think that would be better :)

Reply:
Ah no. This was just curiosity honestly. I hadn't really looked into what `context.WithValue` was doing